arch/arc/core/fault_s.S

```diff
@@ -236,7 +238,7 @@ _do_non_syscall_trap:
 	/* save caller saved registers */
 	_create_irq_stack_frame
 
-#ifdef CONFIG_ARC_HAS_SECURE
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
```

Comment:
I'd suggest to leave this change to a separate commit as it has nothing to do with that particular fix.

Reply:
agreed.